### DISCUSSION TITLE
Expose metrics to a service like prometheus

### DISCUSSION THREAD

**Discussion by networkException** [DNSCrypt/dnscrypt-proxy#1881] [Ideas]
I am currently using [mtail](https://github.com/google/mtail) based on [this reddit post](https://www.reddit.com/r/dnscrypt/comments/kdtkj4/dnscryptproxy_logs_to_prometheus/) to parse logs and generate metrics for prometheus.

Enabling logging however stores queried addresses on disk which I'd like to avoid. 

My idea would be to integrate metrics into dnscrypt-proxy directly, this would also allow for more rich and accurate results.

**Comment by jedisct1** [comment]
According to the documentation, logs can be sent to pipes for that use case: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Logging#custom-log-format--log-processors

**Reply by networkException** [reply]
Alright thank you
